bftengine/src/bftengine/ReplicaBase.hpp

```diff
@@ -12,7 +12,7 @@
 #pragma once
 
 #include <memory>
-
+#include "Replica.hpp"
```

Comment:
This seems wrong. Why would a ReplicaBase include a Replica? Shouldn't it be the other way around?
It looks like you need `IRequestsHandler`. Why don't we introduce an `IRequestsHandlers.hpp` instead?